packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -0,0 +1,111 @@
+import {
+    CheckoutService,
+    Extension,
+    ExtensionCommandType,
+    ExtensionRegion,
+} from '@bigcommerce/checkout-sdk';
+
+export class ExtensionService {
+    private _handlerRemovers: { [extensionId: string]: Array<() => void> } = {};
+
+    constructor(
+        private _checkoutService: CheckoutService,
+        private _setIsShowLoadingIndicator: (show: boolean) => void,
+    ) {}
+
+    async loadExtensions(): Promise<void> {
+        await this._checkoutService.loadExtensions();
+    }
+
+    async renderExtension(container: string, region: ExtensionRegion): Promise<void> {
+        const extension = this._getExtensionByRegion(region);
+
+        if (!extension) {
+            return;
+        }
+
+        await this._checkoutService.renderExtension(container, region);
+
+        if (!this._handlerRemovers[extension.id]) {
+            this._handlerRemovers[extension.id] = [];
+        }
+
+        this._handlerRemovers[extension.id].push(
```

Comment:
Do we need to call these `_handlerRemovers` since that name seems a bit confusing. Can we just call them `_handlers`?

Reply:
`this._handlerRemovers` holds functions to remove an extension's handlers.
It is essential to be invoked after removing each extension from UI, otherwise, we'll handle commands multiple times if an extension is rendered serveral times.

Reply:
I think these are still handlers in concept, the function as part of clean up can be called removehandlers 🤔 I say so because handlerRemovers feels as an odd name and in theory it is a list of handlers/functions to be executed on receiving a command.

Reply:
Renamed.